### ISSUE TITLE
Tooltip suggestions don't show up.

### ISSUE THREAD

**Issue by SlippyDong** [microsoft/pylance-release#4213]
Hi,

I previously had tooltip suggestions that showed me the available options I could enter, but for some reason it stopped working:
![image](https://user-images.githubusercontent.com/71851581/230601210-dfa5629e-686d-4b4e-865f-14a55ac8726d.png)

I've uninstalled vscode, deleted AppData\Roaming\Code and %userprofile%\.vscode, reinstalled vscode, reinstalled python and pylance, but the suggestions still don't show as they did previously.

When I disable Pylance, tooltip suggestions work as expected:
![image](https://user-images.githubusercontent.com/71851581/230602457-61002d5a-e90c-4d85-8267-0aaf8a376fc5.png)

There's nothing exotic in the settings.json (profile) file (which is the selected profile - the settings.json for user and workspace are both empty).
![image](https://user-images.githubusercontent.com/71851581/230602664-7319f828-2699-4890-9f66-d2b47048769c.png)

Can you help me identify what's causing this? I've spent the whole day trying to enable/disable various settings to get this to work again, but I'm stuck.

**Comment by debonte** [comment]
It looks like in the "disable Pylance" screenshot you're using Jedi? I don't think Pylance's hover tooltip has ever shown the value that a `list` was initially set to, even for a constant list.

When you summon this tooltip, what information are you trying to obtain? Maybe answering that question would help us steer you in the right direction.

For example, if you want this information in the tooltip to remind yourself of the names of your dictionary keys, you might consider changing `question_data` to be a `list` of `TypedDict` instead of a `list[dict[str, str]]`. Then when you type `[` to access a key on the dictionary, Pylance will show you the list of supported keys (see example below). For what it's worth, using `TypedDict` would also allow you to use `bool` for `answer` and enable Pylance to validate your key names.

![image](https://user-images.githubusercontent.com/12820898/230663876-4c90bb53-0fa1-43a9-9800-cf6aea5c823a.png)

**Comment by SlippyDong** [comment]
Hi Erik,

Thanks for your reply and suggestion about TypedDict.

To elaborate: I was playing around with vscode - as I just used Pycharm before - and was really pleased to see a complete overview of the entire list with dictionary in the suggestions tooltip window. Since it showed Pylance in that popup I initially assumed this content was suggested by Pylance.

I wanted to see if I could get these suggestions (dictionary/list) pretty printed in the tooltip and added a few lines to the settings.json profile file to try this out. After that, the dictionary and list content then didn't show up anymore, so I removed those lines again, but this content still wouldn't show up again after that. I went on a search to see if I could get it to show again and then found that if I just disabled Pylance, I would get the content to show up again. (You're correct about Jedi, that's the default fallback language server). I then read somewhere on a forum someone else had encountered a similar issue and installed Pyright instead of Pylance, so I gave that a try and that also populates the tooltip popup window with the available content from the list.

Now, initially I thought that maybe I'd just changed a setting somewhere and messed it up, so I decided to do a complete wipe and reinstall of VScode and Python. Unfortunately, this didn't help so I decided to ask for some help from you. I've also asked a friend to install vscode on his pc and set up python to see if the default indeed does show these suggestions in the tooltip popup, but on his pc the tooltip popup wouldn't show the list suggestions either when Pylance was selected as the language server, so I assume this is the default.

In the meantime I kept looking to see what could "fix" this issue and I realized it could be an editor setting as well, as the editor has several options for showing tooltip suggestions too, but so far I haven't been able to find a setting that enables these suggestions when Pylance is enabled.

At this point it looks like Pylance suppresses these suggestions (which actually seem to come from the editor) when it's activated and there's no option to change this behavior. As it stands, this is the tooltip I get when Pylance is active:

[cid:5c2418b9-bbd8-4bdb-9d00-63c350151ac8]

And this is the tooltip when Pyright is active:
[cid:ff13a3a1-217b-4f37-8997-3b84a62f3e82]

You can see that underlined error for question_data doesn't appear when Pylance is active and doesn't show suggestions for the available content from the list, whereas this does show up when Pyright (or Jedi for that matter).

If you know how I can get these suggestions to show up with Pylance active, I'd love to know how to do that. If you would also know how I can get a pretty-printed list to show, instead of this compacted overview, that would be a bonus.

Either way, thans for your help.


________________________________

**Comment by debonte** [comment]
VS Code allows multiple language servers for a particular language to run simultaneously. And when this happens, it will concatenate the information that it receives in response to hover (tooltip) requests from the active language servers.

In your most recent Pyright screenshot, you appear to be running Pyright and Jedi in parallel. Jedi is providing the `SyntaxError` and `question_data = [ ...` sections of the hover data, whereas Pyright is providing the `Expected index or slice expression` diagnostic.

I don't think that running Pyright and Jedi in parallel is common. Should you want to, you can disable Jedi and use Pyright alone by setting the `python.languageServer` config setting to `None`. Or you can use Jedi alone by setting `python.languageServer` to `Jedi` and disabling/uninstalling Pyright. Note that if `python.languageServer` is set to `Default` (which is the default value), it will launch Jedi unless you have Pylance installed, in which case it launches Pylance. 

![image](https://user-images.githubusercontent.com/12820898/231009006-9dfad67f-68da-448c-93a2-34b922dcc28b.png)

> Since it showed Pylance in that popup I initially assumed this content was suggested by Pylance.

It shouldn't be possible for the Python extension to start both Pylance and Jedi, but I wonder if when you saw this behavior you had somehow hit a bug that resulted in both of them being active.

In any case, it sounds like you want to continue using Jedi to get its hover behavior. If you want to request enhancements to Jedi (ex. the pretty printing that you mentioned above), you can file issues in their tracker: https://github.com/pappasam/jedi-language-server/issues.

**Comment by SlippyDong** [comment]
Thank you Erik, this clarifies a lot.  I didn't know Pylance would override the default setting and I also didn't know it was possible to use multiple language servers concurrently. This would also explain duplicate suggestions in the dropdown suggestions. I had indeed left the language server setting to default and perhaps it was possible to see Pylance at the same time as Jedi when I installed and activated Pylance. Can't remember if I reloaded or restarted vscode after this. 

I do like the extended Jedi suggestions and perhaps it's a nice feature to add to Pylance as well. For the moment I'm using [Beauty-Print](https://pypi.org/project/beauty-print/) as a workaround but in  any case, thanks a lot!
